### PR TITLE
ref(postgres): allow users to configure postgres to read from S3

### PR DESCRIPTION
**Pull request by kmala**
Allow postgres to support S3 as storage.

### CODE REVIEW

File: rootfs/docker-entrypoint-initdb.d/001_setup_envdir.sh [hunk 1]
```diff
@@ -3,17 +3,28 @@
 cd "$WALE_ENVDIR"
 
 # access-key-id and access-secret-key files are mounted in via kubernetes secrets
-AWS_ACCESS_KEY_ID=$(cat access-key-id)
-AWS_SECRET_ACCESS_KEY=$(cat access-secret-key)
+if [ "$DATABASE_STORAGE" == "s3" ]; then
+  AWS_ACCESS_KEY_ID=$(cat /var/run/secrets/deis/objectstore/creds/accesskey)
+  AWS_SECRET_ACCESS_KEY=$(cat /var/run/secrets/deis/objectstore/creds/secretkey)
+  AWS_DEFAULT_REGION=$(cat /var/run/secrets/deis/objectstore/creds/region)
```
Comment (bacongobbler) [inline rootfs/docker-entrypoint-initdb.d/001_setup_envdir.sh:9]: this is never written to disk, so we should probably remove it.

File: rootfs/docker-entrypoint-initdb.d/001_setup_envdir.sh [hunk 2]
```diff
@@ -28,12 +39,19 @@ EOF
 # HACK (bacongobbler): minio *must* use us-east-1 and signature version 4
 # for authentication.
 # see https://github.com/minio/minio#how-to-use-aws-cli-with-minio
+if [ "$DATABASE_STORAGE" == "s3" ]; then
 cat << EOF > /root/.aws/config
 [default]
-region = us-east-1
+region = $AWS_DEFAULT_REGION
```
Comment (bacongobbler) [inline rootfs/docker-entrypoint-initdb.d/001_setup_envdir.sh:45]: Ah, I see that it's used here. nvm!

**Comment by bacongobbler** [comment]
code LGTM. Have we tested this against S3 or shall I go ahead and test this with https://github.com/deis/charts/pull/127?

**Comment by kmala** [comment]
I had tested it against s3 and minio.I had created few apps and deleted the database pod and upon starting checked the deis apps:list.

**Comment by bacongobbler** [comment]
All right if you feel confident with the changes then feel free to merge, then wait for the build on https://travis-ci.org/deis/postgres/builds from the merge to finish then go ahead and merge deis/charts#127. @mboersma was burned the last time I pushed a chart change before the image was updated. :)

**Comment by kmala** [comment]
For S3 to work we need fix https://github.com/deis/wal-e/pull/1

**Comment by bacongobbler** [comment]
looks like it's been merged so you should be good to go. :shipit:

**Pull request merged**